code/modules/projectiles/projectile.dm

```diff
@@ -1,353 +1,224 @@
+	var/forcedodge = 0 //to pass through everything
 
-	var/chatlog_attacks = 1
+/obj/item/projectile/New()
+	permutated = list()
+	return ..()
 
-	var/speed = 1 //Amount of deciseconds it takes for projectile to travel. Animation is adjusted accordingly.
-	var/Angle = 0 //For new projectiles
-	var/spread = 0 //Amount of degrees by which the projectiles will be spread DURING MOVEMENT. It exists for chaotic types of projectiles, like bees or something.
-	var/legacy = 0 //Use the legacy projectile system or new pixel movement?
+/obj/item/projectile/proc/Range()
+	range--
+	if(range <= 0 && loc)
+		on_range()
 
-	proc/delete()
-		// Garbage collect the projectiles
-		loc = null
+/obj/item/projectile/proc/on_range() //if we want there to be effects when they reach the end of their range
+	qdel(src)
 
-	proc/Range()
-		if(range)
-			range--
-			if(range <= 0)
-				on_range()
+/obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)
+	if(!isliving(target))
+		return 0
+	var/mob/living/L = target
+	if(blocked < 100) // not completely blocked
+		var/organ_hit_text = ""
+		if(L.has_limbs)
+			organ_hit_text = " in \the [parse_zone(def_zone)]"
+		if(suppressed)
+			playsound(loc, hitsound, 5, 1, -1)
+			to_chat(L, "<span class='userdanger'>You're shot by \a [src][organ_hit_text]!</span>")
 		else
-			return
-
-	proc/on_range() //if we want there to be effects when they reach the end of their range
-		proj_hit = 1
-		del(src)
-
-	proc/on_hit(var/atom/target, var/blocked = 0, var/hit_zone)
-		if(!isliving(target))	return 0
-		if(isanimal(target))	return 0
-		var/mob/living/L = target
-		return L.apply_effects(stun, weaken, paralyze, irradiate, slur, stutter, eyeblur, drowsy, blocked, stamina, jitter)
-
-	proc/OnFired()	//if assigned, allows for code when the projectile gets fired
-		return 1
-
-	proc/check_fire(var/mob/living/target as mob, var/mob/living/user as mob)  //Checks if you can hit them or not.
-		if(!istype(target) || !istype(user))
-			return 0
-		var/obj/item/projectile/test/in_chamber = new /obj/item/projectile/test(get_step_to(user,target)) //Making the test....
-		in_chamber.target = target
-		in_chamber.flags = flags //Set the flags...
-		in_chamber.pass_flags = pass_flags //And the pass flags to that of the real projectile...
-		in_chamber.firer = user
-		var/output = in_chamber.process() //Test it!
-		del(in_chamber) //No need for it anymore
-		return output //Send it back to the gun!
-
-	Bump(atom/A as mob|obj|turf|area)
-		if(A == firer)
+			if(hitsound)
+				var/volume = vol_by_damage()
+				playsound(loc, hitsound, volume, 1, -1)
+			L.visible_message("<span class='danger'>[L] is hit by \a [src][organ_hit_text]!</span>", \
+								"<span class='userdanger'>[L] is hit by \a [src][organ_hit_text]!</span>")	//X has fired Y is now given by the guns so you cant tell who shot you if you could not see the shooter
+		L.apply_effects(stun, weaken, paralyze, irradiate, slur, stutter, eyeblur, drowsy, blocked, stamina, jitter)

```

Comment:
TG's on_hit is used for species-specific projectile handling (like how plant people are affected differently by the floral somatoray). Going to remove this line then.